### PR TITLE
New Dwarf2 plugin that supports x64 binary

### PR DESCRIPTION
**Pull request by HighW4y2H3ll**
The idea of this patch is to add x64 support for taint analysis of DWARF symbols. The `pri_dwarf` plugin is rewritten into `dwarf2` plugin to support the extension of new architectures. Now the `pri_dwarf` plugin is split into two parts: the first part is a Python script that reformats the outputs of `dwarfdump` command into a few JSON files, and the second part is the `dwarf2` plugin which takes in the pre-processed DWARF info.

In conclusion, this patch includes:  
- A Python script that dumps Dwarf info into JSON files. ([dwarfdump.py](https://github.com/panda-re/panda/compare/dev...HighW4y2H3ll:panda:newdwarf?expand=1#diff-82166e850c1f36ef2c46772d45f3e29de0fb2322df2a4ed0a3b267df8e807071))
- `dwarf2` PANDA plugin that replaces the old `pri_dwarf` plugin.
- Fixups in `pri_taint` PANDA plugin to use new `dwarf2` plugin.
- Other minor fixes in `file_taint` and `stringsearch` plugins

### CODE REVIEW

File: panda/plugins/dwarf2/dwarf2.cpp [hunk 1]
```diff
@@ -0,0 +1,2650 @@
+/* PANDABEGINCOMMENT
+ *
+ * Authors:
+ *  Tim Leek               tleek@ll.mit.edu
```
Comment (lacraig2) [inline panda/plugins/dwarf2/dwarf2.cpp]: I'm aware it is the template, but this should only include folks that contributed to this.

File: panda/plugins/file_taint/file_taint.cpp [hunk 2]
```diff
@@ -128,6 +128,7 @@ void read_return(uint64_t file_id, uint64_t bytes_read,
     ReadKey key;
     std::unique_ptr<OsiProc, decltype(free_osiproc)*> proc { get_current_process(first_cpu), free_osiproc };
     std::unique_ptr<OsiThread, decltype(free_osithread)*> thr { get_current_thread(first_cpu), free_osithread };
+    if (!thr)   return;
```
Comment (lacraig2) [inline panda/plugins/file_taint/file_taint.cpp:131]: Will this work with the unique_ptr?

File: panda/plugins/stringsearch/stringsearch.cpp [hunk 3]
```diff
@@ -24,6 +24,7 @@ PANDAENDCOMMENT */
 #include <sstream>
 #include <string>
 #include <iostream>
+#include <array>
```
Comment (lacraig2) [inline panda/plugins/stringsearch/stringsearch.cpp:27]: Import added, but is it used anywhere?

**Review by lacraig2 — COMMENTED** [review]
@HighW4y2H3ll you should rebase this on `dev` to get the CI to pass. (e.g. `git rebase -i origin/dev`)

This is due to no fault of your own, but it is necessary.

File: panda/plugins/file_taint/file_taint.cpp (hunk 2 above)
Comment (HighW4y2H3ll) [reply panda/plugins/file_taint/file_taint.cpp:131]: For the nullptr check, yup, it worked in my branch which was based on an older version of PANDA. unique_ptr overloads `operator bool()`, so I believe it should also work here.

File: panda/plugins/stringsearch/stringsearch.cpp (hunk 3 above)
Comment (HighW4y2H3ll) [reply panda/plugins/stringsearch/stringsearch.cpp:27]: yup, I got a compiler error at these 2 lines: https://github.com/panda-re/panda/blob/0afc7c38ff454a215a475d91e2ce625f99bdee68/panda/plugins/stringsearch/stringsearch.cpp#L60-L61

File: panda/plugins/dwarf2/dwarf2.cpp (hunk 1 above)
Comment (HighW4y2H3ll) [reply panda/plugins/dwarf2/dwarf2.cpp]: Updated!

**Comment by lacraig2** [comment]
This PR isn't passing tests for reasons unrelated to it.

It does pass `test_installer` which indicates that it all compiles.

I will be merging this without tests if no one objects within a day or so.

**Pull request merged**